asf_core_data/pipeline/preprocessing/data_cleaning.py

```diff
@@ -486,11 +483,18 @@ def custom_clean_features(df):
     df = clean_EFF_SCORES(df)
     # [Additional cleaning functions here]
 
-    # Cap features
-    for feat in df.columns:
-        if feat in cap_value_dict.keys():
-            cap_value = cap_value_dict[feat]
-            df = cap_feature_values(df, feat, cap_value=cap_value)
+    if cap_features:
+
+        cap_value_dict = {
+            "NUMBER_HABITABLE_ROOMS": 10,
+            "NUMBER_HEATED_ROOMS": 10,
+        }
+
+        # Cap features
+        for feat in df.columns:
+            if feat in cap_value_dict.keys():
+                cap_value = cap_value_dict[feat]
+                df = cap_feature_values(df, feat, cap_value=cap_value)
```

Comment:
can't we loop through the cap_value_dict instead of all variables?
```suggestion
        for feat in cap_value_dict.keys():
            df = cap_feature_values(df, feat, cap_value=cap_value_dict[feat])
```

Reply:
We'd still need to check whether whether the feature is in the loaded columns, so either way we have to iterate over both lists. But I can simplify by calling the dict directly.